Sources/AudioKitUI/Visualizations/AudioFileWaveform.swift

```diff
@@ -0,0 +1,58 @@
+// Copyright AudioKit. All Rights Reserved. Revision History at http://github.com/AudioKit/AudioKitUI/
+
+import SwiftUI
+import AudioKit
+import Accelerate
+
+class AudioFileWaveformViewModel: ObservableObject {
+    @Published var rmsValues = [Float]()
+    var rmsWindowSize: Double
+    
+    init(url: URL, rmsWindowSize: Double) {
+        self.rmsWindowSize = rmsWindowSize
+        rmsValues = getRMSValues(url: url)
+    }
+
+    func getRMSValues(url: URL) -> [Float] {
+        if let audioInformation = loadAudioSignal(audioURL: url) {
+            let signal = audioInformation.signal
+            return createRMSAnalysisArray(signal: signal, windowSize: Int(audioInformation.rate/rmsWindowSize))
+        }
+        return []
+    }
+
+    func createRMSAnalysisArray(signal: [Float], windowSize: Int) -> [Float] {
+        let numberOfSamples = signal.count
+        let numberOfOutputArrays = numberOfSamples / windowSize
+        var outputArray: [Float] = []
+        for index in 0...numberOfOutputArrays-1 {
+            let startIndex = index * windowSize
+            let endIndex = startIndex + windowSize >= signal.count ? signal.count-1 : startIndex + windowSize
+            let arrayToAnalyze = Array(signal[startIndex..<endIndex])
+            var rms: Float = 0
+            vDSP_rmsqv(arrayToAnalyze, 1, &rms, UInt(windowSize))
+            outputArray.append(rms)
+        }
+        return outputArray
+    }
+}
+
+public struct AudioFileWaveform: View {
+    @ObservedObject var viewModel: AudioFileWaveformViewModel
+
+    public init(url: URL, rmsWindowSize: Double = 256) {
+        viewModel = AudioFileWaveformViewModel(url: url,
+                                               rmsWindowSize: rmsWindowSize)
+    }
+
+    public var body: some View {
+        AudioWaveform(rmsVals: viewModel.rmsValues)
+            .fill(Color.gray)
```

Comment:
Is there a recommended way to override this? Or is it be best for one to write their own implementation? 

Perhaps by adding an argument with a default of `gray`?

either way, thx!

Reply:
https://github.com/AudioKit/AudioKitUI/pull/57

Reply:
NOICE! Thank you :)